moonraker/components/authorization.py

```diff
@@ -356,16 +395,66 @@ async def _handle_password_reset(self,
             'action': "user_password_reset"
         }
 
-    def _login_jwt_user(self,
-                        web_request: WebRequest,
-                        create: bool = False
-                        ) -> Dict[str, Any]:
+    async def _login_ldap_user(self, username, password) -> bool:
+        if self.ldap_server is None or self.ldap_base_dn is None \
+                or self.ldap_group_dn is None \
+                or self.ldap_bind_password is None \
+                or self.ldap_bind_dn is None:
+            raise self.server.error(
+                "ldap: Configuration is not given", 401
+            )
+        base_dn = str(self.ldap_base_dn)
+        client = LDAPClient(self._generate_ldap_url_(str(self.ldap_server)))
+        client.set_credentials("SIMPLE", self.ldap_bind_dn,
+                               self.ldap_bind_password)
+        client.set_cert_policy("allow")
+        bind_success = False
+        try:
+            async with client.connect(is_async=True, timeout=10) as conn:
+                ldap_filter = ("(&(objectClass=Person)(%s=" + '%s))') % \
+                              ("sAMAccountName"
+                               if self.ldap_type_ad
+                               else "uid",
+                               username)
+                user = await conn.search(
+                    base_dn,
+                    LDAPSearchScope.SUBTREE,
+                    ldap_filter,
+                    ['memberOf']
+                )
+                conn.close()
```

Comment:
There is no need to call `close()`, the context manager will close the connection once it exits.